### PR TITLE
Add confirm dialog before Permanently delete

### LINKED ISSUE #60383: Data Views: Pages: Improve clarity of the Trash and Permanently delete flows
### Description

See previous similar issue https://github.com/WordPress/gutenberg/issues/52639

From the Site editor > pages, a page can be 'moved to the trash'. Screenshot:

However, the confirm dialog that opens when trashing a page, asks users whether they want to _delete_ the page. Screenshot:

That's less than ideal. It's confusing for users, as they will not trust the editor and potentially fear the page is going to be _permantently deleted_.

After confirming the action, a snackbar notice will confirm the page has been 'moved to the trash'. Screenshot:

Once the page is in the trash, useres can permanently delete it. One of the available actions is 'Permanently delete'. Screenshot:

After clicking Permanently delete' there is no confirmation step. The page is _permanently deleted_ right away. A snackbar notice will confirm the permanent deletion:

Overall, I think this flow is unclear and confusing for users.
- I'm not sure whether the 'move to trash' action should go through a confirm dialog in the first place. That's not a destructive action. Anyways, the confirm dialog copy and the button text are wrong because they refer to 'delete'.
- The permanent deletion action doesn't go through a confirm step. This is a destructive action. I'd consider to add a confirm dialog to avoid data loss by accident.

## Some history
Traditionally, the classic admin never implemented confirmation dialogs for the 'trash page' and 'delete permanently page' actions. The reasoning was that the first action isn't a destructive one. The second action is desctructive but users are in the context of hte 'trash' and it's clear enough that deelting from the trash _deletes permanently_.

Regardless od the historical approach illustrated above, it is worth considering whether it would be beneficial to think at a better flow. My 2 cents:
- The trash action shouldn't have a confirm dialog.
- The permantently delete action should have it.

### Step-by-step reproduction instructions

- Go to the site editor > Pages > switch to the List view to make things more clear.
- Trash a page.
- Observe the confirmation dialog says 'delete'.
- Once the page is trashed, go to the Pages Trash section
- In the Actione menu for a page, click 'Delete permanently'.
- Observe there is no confirmation step.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by karthick-murugan**
## What?
Fixes https://github.com/WordPress/gutenberg/issues/60383, where the "Permanently Delete" action for trashed posts in the WordPress editor was not functioning as expected. This PR improves the logic and introduces a confirmation modal to ensure users can confirm their actions before permanently deleting posts.

## Why?
The current implementation of the "Permanently Delete" action did not provide a clear user experience and lacked safeguards like a confirmation modal. This could lead to unintentional deletions. By addressing this issue, the PR ensures a better and more accessible user experience while reducing the likelihood of errors.

## How?
Added a modal confirmation step to confirm the deletion of selected posts.

## Testing Instructions

- Open the WordPress admin dashboard.
- Navigate to any list of posts.
- Move one or more posts to the trash.
- Select the trashed posts and use the "Permanently Delete" option.
- Confirm the deletion in the modal dialog.
- Observe the success or error notices for both individual and bulk deletion scenarios.

## Screenshots or screencast 

https://github.com/user-attachments/assets/09a09a0c-0d0f-417a-ac78-c69db68febf4

### CODE REVIEW

**Comment by Mamaduka** [comment]
@karthick-murugan, entity actions already have built-in support for confirmation modals. See [`delete-post`](https://github.com/WordPress/gutenberg/blob/b66f9ff095f5b70f9af15a50af1772c0e201a5e6/packages/fields/src/actions/delete-post.tsx#L50) and [`trash-post`](https://github.com/WordPress/gutenberg/blob/b66f9ff095f5b70f9af15a50af1772c0e201a5e6/packages/fields/src/actions/trash-post.tsx#L42) actions for examples.

File: packages/fields/src/actions/permanently-delete-post.tsx [hunk 1]
```diff
@@ -25,93 +34,153 @@ const permanentlyDeletePost: Action< PostWithPermissions > = {
...
+		const { deleteEntityRecord } = useDispatch( coreStore );
+
+		return (
+			<VStack spacing="5">
+				<Text>
+					{ items.length > 1
+						? sprintf(
+								// translators: %d: number of items to delete.
+								_n(
+									'Permanently delete %d item?',
```
Comment (ntsekouras) [inline packages/fields/src/actions/permanently-delete-post.tsx]: Can we update the copy here to match other similar actions that start with: `Are you sure you want to...`?

**Review by ntsekouras — COMMENTED** [review]
Thanks for the PR! This looks good code wise. I've left a comment regarding the copy. Also it would be good to rebase.

**Review by ntsekouras — APPROVED** [review]
Looks good, thank you!

**Pull request merged**